### PR TITLE
[llama-mm] Make text decoder exportable

### PR DESCRIPTION
**Pull request by larryliu0820**
Summary: Adds source transformation and changes example input to make text decoder exportable.

Test Plan: Added new unit test.

Reviewers:

Subscribers:

Tasks:

Tags:

### Summary
[PLEASE REMOVE] See [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests) for ExecuTorch PR guidelines.

[PLEASE REMOVE] If this PR closes an issue, please add a `Fixes #<issue-id>` line.

[PLEASE REMOVE] If this PR introduces a fix or feature that should be the upcoming release notes, please add a "Release notes: <area>" label. For a list of available release notes labels, check out [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests).

### Test plan
[PLEASE REMOVE] How did you test this PR? Please write down any manual commands you used and note down tests that you have written if applicable.

### CODE REVIEW

File: examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import unittest
+
+import torch
+
+from executorch.examples.models.llama3_2_vision.text_decoder.model import (
+    Llama3_2Decoder,
+)
+from torch.testing import assert_close
+
+params = {
```
Comment (dvorjackz) [inline examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py:21]: Can load https://github.com/pytorch/executorch/blob/main/examples/models/llama3_2_vision/text_decoder/params/demo_config.json, also can set n_layers to 1 to make test go faster

File: examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (dvorjackz) [inline examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py:1]: Can rename to `test_aoti.py` or something

File: examples/models/llama3_2_vision/text_decoder/model.py [hunk 3]
```diff
@@ -147,39 +151,46 @@ def __init__(self, **kwargs):
             self.model_.setup_caches(
                 batch_size=1,
                 dtype=self.dtype,
+                encoder_max_seq_len=self.encoder_max_seq_len,
                 decoder_max_seq_len=self.max_seq_len,
             )
+        # number of tokens for example input
+        self.n_tokens = 34
```
Comment (dvorjackz) [inline examples/models/llama3_2_vision/text_decoder/model.py:158]: Why 34 just curious

File: examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        for param in model.parameters():
+            param.requires_grad = False
+        for child in model.children():
+            self._set_requires_grad_false(child)
+
+    def test_llama3_2_text_decoder(self) -> None:
+        with tempfile.NamedTemporaryFile(mode="w") as param_file:
+            json.dump(params, param_file, indent=2)
+            param_file.flush()
+            print(param_file.name)
```
Comment (dvorjackz) [inline examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py:55]: rm

File: examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                dynamic_shapes=model.get_dynamic_shapes(),
+            )
+        with tempfile.TemporaryDirectory() as tmpdir:
+            path = torch._inductor.aoti_compile_and_package(
+                ep,
+                model.get_example_inputs(),
+                kwargs=model.get_example_kwarg_inputs(),
+                package_path=os.path.join(tmpdir, "text_decoder.pt2"),
+            )
+            print(path)
```
Comment (dvorjackz) [inline examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py:82]: rm

File: examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py (hunk 1 above)
Comment (larryliu0820) [reply examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py:21]: Changed, reading json is not very convenient in unit test

File: examples/models/llama3_2_vision/text_decoder/model.py (hunk 3 above)
Comment (larryliu0820) [reply examples/models/llama3_2_vision/text_decoder/model.py:158]: Just a random number

**Review by jackzhxng — APPROVED** [review]
